src/component/public.ts

```diff
@@ -62,23 +64,26 @@ export const replace = mutation({
     newKey: v.any(),
     value: v.any(),
     summand: v.optional(v.number()),
+    namespace: v.optional(v.any()),
+    newNamespace: v.optional(v.any()),
   },
   returns: v.null(),
   handler: async (ctx, args) => {
-    await deleteHandler(ctx, { key: args.currentKey });
+    await deleteHandler(ctx, { key: args.currentKey, namespace: args.namespace });
     await insertHandler(ctx, {
       key: args.newKey,
       value: args.value,
       summand: args.summand,
+      namespace: args.newNamespace,
```

Comment:
we still need to add the optimization here to short-circuit if the value hasn't changed, ya? Or is that done at a different layer?

Reply:
yes that optimization hasn't been implemented yet. namespaces don't really change the problem, so i'll put that off to another PR